scheduler/src/cook/mesos/util.clj

```diff
@@ -438,3 +439,32 @@
   [instance]
   (some #{(:instance/status instance)} #{:instance.status/running
                                          :instance.status/unknown}))
+
+(defn close-when-ch!
+  "When the value passed in is a channel, close it. Otherwise, do nothing"
+  [maybe-ch]
+  (try
+    (async/close! maybe-ch)
+    (catch Exception e)))
+
+(defn chime-at-ch
```

Comment:
Naming is more @sradack 's forte.

If we are calling `f` with current time, we could call the function `process-heartbeat`.